src/test/java/pl/agasior/interviewprep/services/question/QuestionReaderTest.java

```diff
@@ -0,0 +1,96 @@
+package pl.agasior.interviewprep.services.question;
+
+import org.junit.jupiter.api.*;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.test.context.ActiveProfiles;
+import org.springframework.test.context.junit.jupiter.SpringExtension;
+import org.springframework.test.web.servlet.MockMvc;
+import org.springframework.test.web.servlet.setup.MockMvcBuilders;
+import org.springframework.web.context.WebApplicationContext;
+import pl.agasior.interviewprep.dto.CreateQuestionRequest;
+import pl.agasior.interviewprep.dto.GetQuestionsByTagsRequest;
+import pl.agasior.interviewprep.dto.UserDto;
+import pl.agasior.interviewprep.entities.Role;
+import pl.agasior.interviewprep.repositories.QuestionRepository;
+import pl.agasior.interviewprep.testutils.DatabasePreparer;
+import pl.agasior.interviewprep.testutils.FakeUserReader;
+import pl.agasior.interviewprep.testutils.RequestFactory;
+
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.Set;
+
+import static org.springframework.test.web.servlet.result.MockMvcResultHandlers.print;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.status;
+
+@ExtendWith(SpringExtension.class)
+@SpringBootTest
+@ActiveProfiles("test")
+@TestInstance(TestInstance.Lifecycle.PER_CLASS)
+public class QuestionReaderTest {
+    @Autowired
+    private RequestFactory requestFactory;
+
+    @Autowired
+    private DatabasePreparer databasePreparer;
+
+    @Autowired
+    private QuestionRepository questionRepository;
+
+    @Autowired
+    private FakeUserReader fakeUserReader;
+
+    private final MockMvc mockMvc;
+
+    QuestionReaderTest(WebApplicationContext webApplicationContext) {
+        this.mockMvc = MockMvcBuilders.webAppContextSetup(webApplicationContext).build();
+    }
+
+    @BeforeEach
+    @AfterAll
+    void clearDatabase() {
+        databasePreparer.clear();
+    }
+
+    @Nested
+    class ReadQuestion {
+
+        @Test
+        void findQuestionsByTags() throws Exception {
+            final var user = UserDto.builder().username("admin").roles(Set.of(Role.Admin, Role.User)).build();
```

Comment:
User creation isn't really part of this test and I bet its could be reused somewhere - maybe create factory class that will do that?